app/src/main/kotlin/com/simplemobiletools/calculator/helpers/GetGraphTask.kt

```diff
@@ -11,36 +12,52 @@ import java.net.URL
 
 class GetGraphTask(val points: String, val drawActivity: DrawActivity) : AsyncTask<Void, Void, String>() {
 
+    @SuppressLint("WrongConstant")
     override fun doInBackground(vararg params: Void?): String? {
-        val url = URL("http://159.203.16.16/equation")
+        try {
+            val url = URL("http://159.203.16.16/equation")
 
-        val httpClient = url.openConnection() as HttpURLConnection
+            val httpClient = url.openConnection() as HttpURLConnection
+            httpClient.requestMethod = "POST"
+            httpClient.setRequestProperty("Content-Type", "application/x-www-form-urlencoded")
+            httpClient.setRequestProperty("charset", "utf-8")
+            httpClient.useCaches = false
 
-        httpClient.requestMethod = "POST"
-        httpClient.setRequestProperty( "Content-Type", "application/x-www-form-urlencoded")
-        httpClient.setRequestProperty( "charset", "utf-8")
-        httpClient.useCaches = false
+            val writer = OutputStreamWriter(httpClient.outputStream)
+            writer.write("strokes=" + points)
+            writer.flush()
 
-        val writer = OutputStreamWriter(httpClient.outputStream)
-        writer.write("strokes=" + points)
-        writer.flush()
-
-        if (httpClient.responseCode == HttpURLConnection.HTTP_OK) {
-            try {
+            var data: String
+            if (httpClient.responseCode == 200) {
                 val stream = BufferedInputStream(httpClient.inputStream)
-                val data: String = readStream(inputStream = stream)
+                data = readStream(inputStream = stream)
+                httpClient.disconnect()
                 return data
-            } catch (e: Exception) {
-                e.printStackTrace()
-            } finally {
+            } else if (httpClient.responseCode == 404) {
                 httpClient.disconnect()
+                pingWithError("Page not found 404!", "There has been a connection problem, returning you to the main page")
+                killThread()
+            } else if (httpClient.responseCode == 400) {
```

Comment:
Could handle all other responseCodes in an else block.

Reply:
the server only sends 400 and 404. Maybe if it really fails hard we could get a 500. An `else` would be perfect I think.